website/content/docs/api-clients/api.mdx

```diff
@@ -79,3 +81,197 @@ The following method conventions are used within Boundary's API:
+The controllers may either run out of resources or overwhelm the database server and exhaust its resources.
+API rate limiting lets you configure limits on the rate of API requests to help manage your resources and prevent them from being overwhelmed.
+
+### Quotas
+
+Boundary creates quotas to track the number of requests in a given time period.
+By default, Boundary tracks the number of requests by auth token, IP address, and the overall total.
+Boundary reserves a portion of its memory for tracking quotas to ensure that it does not consume too much memory if there is a sudden burst of requests.
+
+If Boundary is unable to store a quota, it limits the request with a 503 HTTP status code.
+You can configure the maximum number of quotas Boundary allows using the `api_rate_limit_max_quotas` variable.
+There are also two metrics that allow you to monitor quota tracking:
+
+- `boundary_controller_api_ratelimiter_quota_stroage_capacity`
+- `boundary_controller_api_ratelimiter_quota_storage_usage`
+
+### Default limits
+
+API rate limiting is enforced on the controllers.
+There are separate configurable limits for each combination of resource and action.
+By default, the limits for `list` actions are:
+
+- 150 requests per 30 seconds per auth token
+- 1,500 requests per 30 seconds per IP address
+- 1,500 requests per 30 seconds in total
+
+The default limits for all other actions are:
+
+- 3,000 requests per 30 seconds per auth token
+- 30,000 requests per 30 seconds per IP address
+- 30,000 requests per 30 seconds in total
+
+You can override the default settings and configure other specific limitations using the `api_rate_limit` stanza in the controller configuration.
+
+### HTTP headers
+
+Clients that make requests to the controller API can inspect HTTP response headers to understand the configured limits and current usage.
+Each response contains the following headers:
+
+- `RateLimit` - Provides the current limit, number of remaining requests, and the time at which the quota will reset for the limit that is closest to being exhausted for the requested resource and action.
+- `RateLimit-Policy` - Describes the limits for the requested resource and action.
+
+If the request is limited, Boundary sends the client a 429 HTTP status code with a `Retry-After` header.
+The `Retry-After` header contains the number of seconds the client should wait before it sends the request again.
+
+### More information
+
+Refer to the [controller stanza](/boundary/docs/configuration/controller#api_rate_limit) documentation for the specific `api_rate_limit` configuration options.
+Some example configurations are listed below.
+
+### Rate limiting configuration examples
+
+The following example shows a simple configuration where the same limits are applied to all resources and examples:
+
+```hcl
+controller {
+  api_rate_limit {
+	resources = ["*"]
+	actions   = ["*"]
+	per  	= "total"
+	limit 	= 500
+	period	= "1s"
```

Comment:
> Is there something funny about the indentation here? Are there tabs here instead of spaces? What's the standard?

I think I copied/pasted these examples from the RFC @moduli , so it's very possible the indentation got messed up. I will take a look and compare to existing code blocks.

Reply:
Interesting. I guess it looks fine on the site compared to the code. I'm guessing it's a tabs/spaces situation. Not a big deal if it looks fine on the site.

Just some example comparisons in this PR for reference:
![Screenshot 2023-12-08 at 2 59 50 PM](https://github.com/hashicorp/boundary/assets/2474253/de12094d-4517-4155-8d00-6f097559dda5)
![Screenshot 2023-12-08 at 3 00 01 PM](https://github.com/hashicorp/boundary/assets/2474253/fb0d104e-9e9a-41c1-bcf6-4f08bbfae679)

Reply:
okay I think that should do it, thanks for the catch @moduli